src/fillalgebra.jl

```diff
@@ -261,6 +261,33 @@ end
 -(a::AbstractRange{T}, b::Zeros{V, 1}) where {T, V} = a + b
 
 
+function +(a::Zeros{T, 2}, b::UniformScaling{V}) where {T, V<:Number}
+    n = LinearAlgebra.checksquare(a)
+    return Diagonal(Fill{promote_type(T,V)}(b.λ, n))
+end
+function +(a::RectOrDiagonal{T, <:AbstractFill{T, 1}}, b::UniformScaling) where {T}
+    LinearAlgebra.checksquare(a)
+    return Diagonal(a.diag .+ b.λ)
+end
```

Comment:
The reverse order falls back to these definitions: https://github.com/JuliaLang/julia/blob/e3b681c6872d87d74db63dd7ec3da2beee8e3ae4/stdlib/LinearAlgebra/src/uniformscaling.jl#L153